Migrate `anastasis` and `anastasis-gtk`

- [ ] https://github.com/ngi-nix/anastasis
  - https://github.com/ngi-nix/ngipkgs/pull/136
  - [ ] migrate tests: https://github.com/ngi-nix/anastasis/blob/master/flake.nix#L58-L110
- https://github.com/ngi-nix/anastasis-gtk

Tracking issue: https://github.com/ngi-nix/ngipkgs/issues/12.

After bumping `anastasis` from version 0.2.0 to 0.4.0 and changing the fetch URL from https://git.taler.net/anastasis.git instead of `mirror://gnu/anastasis/[...].tar.gz`, we get a cryptic build failure.

<details>
<summary>Build log</summary>
```
❯ nix flake check -L
warning: flake output attribute 'overlay' is deprecated; use 'overlays.default' instead
warning: flake output attribute 'defaultPackage' is deprecated; use 'packages.<system>.default' instead
warning: flake output attribute 'devShell' is deprecated; use 'devShells.<system>.default' instead
anastasis> unpacking sources
anastasis> unpacking source archive /nix/store/s38wf8xcxlygscyp2r02lsvjwdr95r4l-anastasis
anastasis> source root is anastasis
anastasis> patching sources
anastasis> patching script interpreter paths in src/cli
anastasis> src/cli/test_anastasis_reducer_enter_secret.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_anastasis_reducer_recovery_no_pay.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_anastasis_reducer_backup_enter_user_attributes.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/setup.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/sh"
anastasis> src/cli/test_anastasis_reducer_add_authentication.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_anastasis_reducer_initialize_state.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_anastasis_reducer_recovery_enter_user_attributes.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_anastasis_reducer_done_policy_review.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_anastasis_reducer_select_country.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_anastasis_reducer_recovery_hanging.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_anastasis_reducer_select_continent.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_iban.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> src/cli/test_anastasis_reducer_done_authentication.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash"
anastasis> autoreconfPhase
anastasis> autoreconf: export WARNINGS=
anastasis> autoreconf: Entering directory '.'
anastasis> autoreconf: running: autopoint --force
anastasis> Copying file m4/codeset.m4
anastasis> Copying file m4/extern-inline.m4
anastasis> Copying file m4/fcntl-o.m4
anastasis> Copying file m4/glibc2.m4
anastasis> Copying file m4/glibc21.m4
anastasis> Copying file m4/intdiv0.m4
anastasis> Copying file m4/intl.m4
anastasis> Copying file m4/intldir.m4
anastasis> Copying file m4/intlmacosx.m4
anastasis> Copying file m4/intmax.m4
anastasis> Copying file m4/inttypes-pri.m4
anastasis> Copying file m4/inttypes_h.m4
anastasis> Copying file m4/lcmessage.m4
anastasis> Copying file m4/lock.m4
anastasis> Copying file m4/longlong.m4
anastasis> Copying file m4/printf-posix.m4
anastasis> Copying file m4/size_max.m4
anastasis> Copying file m4/stdint_h.m4
anastasis> Copying file m4/threadlib.m4
anastasis> Copying file m4/uintmax_t.m4
anastasis> Copying file m4/visibility.m4
anastasis> Copying file m4/wchar_t.m4
anastasis> Copying file m4/wint_t.m4
anastasis> Copying file m4/xsize.m4
anastasis> Copying file po/Makevars.template
anastasis> autoreconf: running: aclocal --force -I m4
anastasis> autoreconf: configure.ac: tracing
anastasis> autoreconf: running: libtoolize --copy --force
anastasis> libtoolize: putting auxiliary files in '.'.
anastasis> libtoolize: copying file './ltmain.sh'
anastasis> libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
anastasis> libtoolize: copying file 'm4/libtool.m4'
anastasis> libtoolize: copying file 'm4/ltoptions.m4'
anastasis> libtoolize: copying file 'm4/ltsugar.m4'
anastasis> libtoolize: copying file 'm4/ltversion.m4'
anastasis> libtoolize: copying file 'm4/lt~obsolete.m4'
anastasis> autoreconf: configure.ac: not using Intltool
anastasis> autoreconf: configure.ac: not using Gtkdoc
anastasis> autoreconf: running: aclocal --force -I m4
anastasis> autoreconf: running: /nix/store/wafffsakgcxdcfwhrrlylw2wkawvdcww-autoconf-2.71/bin/autoconf --force
anastasis> configure.ac:45: warning: The macro `AC_HELP_STRING' is obsolete.
anastasis> configure.ac:45: You should run autoupdate.
anastasis> ./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
anastasis> m4/libgnurl.m4:38: LIBGNURL_CHECK_CONFIG is expanded from...
anastasis> lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
anastasis> configure.ac:45: the top level
anastasis> configure.ac:45: warning: The macro `AC_HELP_STRING' is obsolete.
anastasis> configure.ac:45: You should run autoupdate.
anastasis> ./lib/autoconf/general.m4:204: AC_HELP_STRING is expanded from...
anastasis> m4/libcurl.m4:38: LIBCURL_CHECK_CONFIG is expanded from...
anastasis> lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
anastasis> configure.ac:45: the top level
anastasis> autoreconf: running: /nix/store/wafffsakgcxdcfwhrrlylw2wkawvdcww-autoconf-2.71/bin/autoheader --force
anastasis> autoreconf: running: automake --add-missing --copy --force-missing
anastasis> configure.ac:19: installing './compile'
anastasis> configure.ac:26: installing './config.guess'
anastasis> configure.ac:26: installing './config.sub'
anastasis> configure.ac:11: installing './install-sh'
anastasis> configure.ac:11: installing './missing'
anastasis> src/authorization/Makefile.am: installing './depcomp'
anastasis> parallel-tests: installing './test-driver'
anastasis> autoreconf: Leaving directory '.'
anastasis> updateAutotoolsGnuConfigScriptsPhase
anastasis> Updating Autotools / GNU config script to a newer upstream version: ./config.sub
anastasis> Updating Autotools / GNU config script to a newer upstream version: ./config.guess
anastasis> configuring
anastasis> fixing libtool script ./ltmain.sh
anastasis> ./configure
anastasis> configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0 --bindir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/bin --sbindir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/sbin --includedir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/include --oldincludedir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/include --mandir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/share/man --infodir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/share/info --docdir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/share/doc/anastasis --libdir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/lib --libexecdir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/libexec --localedir=/nix/store/a4zcrfn3c7j70zhw3xqj3rm99kvh9pxg-anastasis-0.4.0/share/locale --with-gnunet=/nix/store/9n8awg01kwa40m86cxmldwj2m0vwywg8-gnunet-0.20.0 --with-exchange=/nix/store/yxwbpb55704f3j07ql45jqha7qmyx122-taler-exchange-0.9.2 --with-merchant=/nix/store/cmj456glf25r8gx3ar7npami5wg6pvgy-taler-merchant-0.9.2
anastasis> checking for a BSD-compatible install... /nix/store/y9gr7abwxvzcpg5g73vhnx1fpssr5frr-coreutils-9.3/bin/install -c
anastasis> checking whether build environment is sane... yes
anastasis> checking for a race-free mkdir -p... /nix/store/y9gr7abwxvzcpg5g73vhnx1fpssr5frr-coreutils-9.3/bin/mkdir -p
anastasis> checking for gawk... gawk
anastasis> checking whether make sets $(MAKE)... yes
anastasis> checking whether make supports nested variables... yes
anastasis> checking how to create a pax tar archive... gnutar
anastasis> checking whether make supports nested variables... (cached) yes
anastasis> checking for gawk... (cached) gawk
anastasis> checking for gcc... gcc
anastasis> checking whether the C compiler works... yes
anastasis> checking for C compiler default output file name... a.out
anastasis> checking for suffix of executables...
anastasis> checking whether we are cross compiling... no
anastasis> checking for suffix of object files... o
anastasis> checking whether the compiler supports GNU C... yes
anastasis> checking whether gcc accepts -g... yes
anastasis> checking for gcc option to enable C11 features... none needed
anastasis> checking whether gcc understands -c and -o together... yes
anastasis> checking whether make supports the include directive... yes (GNU style)
anastasis> checking dependency style of gcc... none
anastasis> checking for gcc... gcc
anastasis> checking whether the compiler supports GNU Objective C... no
anastasis> checking whether gcc accepts -g... no
anastasis> checking dependency style of gcc... none
anastasis> checking whether ln -s works... yes
anastasis> checking whether make sets $(MAKE)... (cached) yes
anastasis> checking build system type... x86_64-pc-linux-gnu
anastasis> checking host system type... x86_64-pc-linux-gnu
anastasis> checking how to print strings... printf
anastasis> checking for a sed that does not truncate output... /nix/store/x23by79p38ll0js1alifmf3y56vqfs49-gnused-4.9/bin/sed
anastasis> checking for grep that handles long lines and -e... /nix/store/xafzciap7acqhfx84dvqkp18bg4lrai3-gnugrep-3.11/bin/grep
anastasis> checking for egrep... /nix/store/xafzciap7acqhfx84dvqkp18bg4lrai3-gnugrep-3.11/bin/grep -E
anastasis> checking for fgrep... /nix/store/xafzciap7acqhfx84dvqkp18bg4lrai3-gnugrep-3.11/bin/grep -F
anastasis> checking for ld used by gcc... ld
anastasis> checking if the linker (ld) is GNU ld... yes
anastasis> checking for BSD- or MS-compatible name lister (nm)... nm
anastasis> checking the name lister (nm) interface... BSD nm
anastasis> checking the maximum length of command line arguments... 1572864
anastasis> checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
anastasis> checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
anastasis> checking for ld option to reload object files... -r
anastasis> checking for file... file
anastasis> checking for objdump... objdump
anastasis> checking how to recognize dependent libraries... (cached) pass_all
anastasis> checking for dlltool... no
anastasis> checking how to associate runtime and link libraries... printf %s\n
anastasis> checking for archiver @FILE support... @
anastasis> checking for strip... strip
anastasis> checking for ranlib... ranlib
anastasis> checking command to parse nm output from gcc object... ok
anastasis> checking for sysroot... no
anastasis> checking for a working dd... /nix/store/y9gr7abwxvzcpg5g73vhnx1fpssr5frr-coreutils-9.3/bin/dd
anastasis> checking how to truncate binary pipes... /nix/store/y9gr7abwxvzcpg5g73vhnx1fpssr5frr-coreutils-9.3/bin/dd bs=4096 count=1
anastasis> checking for mt... no
anastasis> checking if : is a manifest tool... no
anastasis> checking for stdio.h... yes
anastasis> checking for stdlib.h... yes
anastasis> checking for string.h... yes
anastasis> checking for inttypes.h... yes
anastasis> checking for stdint.h... yes
anastasis> checking for strings.h... yes
anastasis> checking for sys/stat.h... yes
anastasis> checking for sys/types.h... yes
anastasis> checking for unistd.h... yes
anastasis> checking for dlfcn.h... yes
anastasis> checking for objdir... .libs
anastasis> checking if gcc supports -fno-rtti -fno-exceptions... no
anastasis> checking for gcc option to produce PIC... -fPIC -DPIC
anastasis> checking if gcc PIC flag -fPIC -DPIC works... yes
anastasis> checking if gcc static flag -static works... no
anastasis> checking if gcc supports -c -o file.o... yes
anastasis> checking if gcc supports -c -o file.o... (cached) yes
anastasis> checking whether the gcc linker (ld -m elf_x86_64) supports shared libraries... yes
anastasis> checking whether -lc should be explicitly linked in... no
anastasis> checking dynamic linker characteristics... GNU/Linux ld.so
anastasis> checking how to hardcode library paths into programs... immediate
anastasis> checking for shl_load... no
anastasis> checking for shl_load in -ldld... no
anastasis> checking for dlopen... yes
anastasis> checking whether a program can dlopen itself... yes
anastasis> checking whether a statically linked program can dlopen itself... yes
anastasis> checking whether stripping libraries is possible... yes
anastasis> checking if libtool supports shared libraries... yes
anastasis> checking whether to build shared libraries... yes
anastasis> checking whether to build static libraries... no
anastasis> checking for doxygen... no
anastasis> configure: WARNING: doxygen not found - will not generate any doxygen documentation
anastasis> checking for perl... no
anastasis> checking whether to compile documentation ONLY... no
anastasis> checking whether NLS is requested... yes
anastasis> checking for msgfmt... /nix/store/gzk32jyj8r3l4yhjn80gl10p355g3ngi-gettext-0.21.1/bin/msgfmt
anastasis> checking for gmsgfmt... /nix/store/gzk32jyj8r3l4yhjn80gl10p355g3ngi-gettext-0.21.1/bin/msgfmt
anastasis> checking for xgettext... /nix/store/gzk32jyj8r3l4yhjn80gl10p355g3ngi-gettext-0.21.1/bin/xgettext
anastasis> checking for msgmerge... /nix/store/gzk32jyj8r3l4yhjn80gl10p355g3ngi-gettext-0.21.1/bin/msgmerge
anastasis> checking for ld used by gcc... ld -m elf_x86_64
anastasis> checking if the linker (ld -m elf_x86_64) is GNU ld... yes
anastasis> checking for shared library run path origin... done
anastasis> checking how to run the C preprocessor... gcc -E
anastasis> checking for gcc options needed to detect all undeclared functions... none needed
anastasis> checking pkg-config is at least version 0.9.0... yes
anastasis> checking for unsigned long long int... yes
anastasis> checking for long long int... yes
anastasis> checking for gcc... (cached) gcc
anastasis> checking whether the compiler supports GNU C... (cached) yes
anastasis> checking whether gcc accepts -g... (cached) yes
anastasis> checking for gcc option to enable C11 features... (cached) none needed
anastasis> checking whether gcc understands -c and -o together... (cached) yes
anastasis> checking dependency style of gcc... (cached) none
anastasis> checking for CFPreferencesCopyAppValue... no
anastasis> checking for CFLocaleCopyCurrent... no
anastasis> checking for GNU gettext in libc... yes
anastasis> checking whether to use NLS... yes
anastasis> checking where the gettext function comes from... libc
anastasis> checking for stdint.h... (cached) yes
anastasis> checking for stdlib.h... (cached) yes
anastasis> checking for string.h... (cached) yes
anastasis> checking for unistd.h... (cached) yes
anastasis> checking for gawk... (cached) gawk
anastasis> checking for gnurl-config... no
anastasis> checking whether libgnurl is usable... no
anastasis> checking for gawk... (cached) gawk
anastasis> checking for curl-config... /nix/store/r4638iv5a646jz14rl02r6r9cqhggaky-curl-8.3.0-dev/bin/curl-config
anastasis> checking for the version of libcurl... 8.3.0
anastasis> checking for libcurl >= version 7.34.0... yes
anastasis> checking whether libcurl is usable... yes
anastasis> checking for curl_free... yes
anastasis> checking for curl/curl.h... yes
anastasis> checking whether CURLINFO_TLS_SESSION is declared... yes
anastasis> checking for curl_easy_getinfo in -lcurl-gnutls... no
anastasis> checking for curl/curl.h... (cached) yes
anastasis> configure: WARNING: libgnurl not found, trying to use libcurl-gnutls instead.
anastasis> configure: WARNING: GNU Taler requires libcurl-gnutls  >= 7.34
anastasis> checking whether to compile with support for code coverage analysis... no
anastasis> checking for libgnunetutil... given as /nix/store/9n8awg01kwa40m86cxmldwj2m0vwywg8-gnunet-0.20.0
anastasis> checking for gnunet/gnunet_util_lib.h... yes
anastasis> checking for GNUNET_SCHEDULER_run in -lgnunetutil... yes
anastasis> checking pkg-config is at least version 0.9.0... yes
anastasis> checking for the PostgreSQL libraries CPPFLAGS... -I/nix/store/ki3srrjjzqalvh0hd9lmqavp5v9wr9jp-postgresql-14.9/include
anastasis> checking for the PostgreSQL libraries LDFLAGS... -L/nix/store/sk7c7f10fxn7iypqgaaq9l0s9ipkk0mc-postgresql-14.9-lib/lib
anastasis> checking for the PostgreSQL libraries LIBS... -lpq
anastasis> checking for the PostgreSQL version... 14.9
anastasis> checking for libpq-fe.h... yes
anastasis> checking for the PostgreSQL library linking is working... yes
anastasis> checking for sodium.h... yes
anastasis> checking for crypto_core_ed25519_scalar_mul in -lsodium... yes
anastasis> checking for libtalerutil... given as /nix/store/yxwbpb55704f3j07ql45jqha7qmyx122-taler-exchange-0.9.2
anastasis> checking for libgcrypt-config... /nix/store/bf9s0q3wqqfcqmk3335vz0ryqwmaff8f-libgcrypt-1.10.2-dev/bin/libgcrypt-config
anastasis> checking for LIBGCRYPT - version >= 1.6.1... yes (1.10.2)
anastasis> configure: WARNING:
anastasis> ***
anastasis> *** The config script "/nix/store/bf9s0q3wqqfcqmk3335vz0ryqwmaff8f-libgcrypt-1.10.2-dev/bin/libgcrypt-config" was
anastasis> *** built for x86_64-unknown-linux-gnu and thus may not match the
anastasis> *** used host x86_64-pc-linux-gnu.
anastasis> *** You may want to use the configure option --with-libgcrypt-prefix
anastasis> *** to specify a matching config script or use $SYSROOT.
anastasis> ***
anastasis> checking for taler/taler_util.h... yes
anastasis> checking for TALER_b2s in -ltalerutil... yes
anastasis> checking for libtalermerchant... given as /nix/store/cmj456glf25r8gx3ar7npami5wg6pvgy-taler-merchant-0.9.2
anastasis> checking for taler/taler_merchant_service.h... yes
anastasis> checking for TALER_MERCHANT_parse_pay_uri in -ltalermerchant... yes
anastasis> checking for microhttpd... not given
anastasis> checking for MHD_start_daemon in -lmicrohttpd... yes
anastasis> checking for microhttpd.h... yes
anastasis> checking for jansson >= 2.3... yes
anastasis> checking for git... no
anastasis> checking for source being under a VCS... no
anastasis> checking VCS version... "release"
anastasis> checking for pid_t... yes
anastasis> checking for size_t... yes
anastasis> checking for uint16_t... yes
anastasis> checking for uint32_t... yes
anastasis> checking for uint64_t... yes
anastasis> checking for intmax_t... yes
anastasis> checking for uintmax_t... yes
anastasis> checking for strdup... yes
anastasis> checking whether to compile experimental code... no
anastasis> checking that generated files are newer than configure... done
anastasis> configure: creating ./config.status
anastasis> config.status: creating Makefile
anastasis> config.status: creating contrib/Makefile
anastasis> config.status: creating doc/Makefile
anastasis> config.status: creating po/Makefile.in
anastasis> config.status: creating src/Makefile
anastasis> config.status: creating src/authorization/Makefile
anastasis> config.status: creating src/authorization/libanastasiseufin/Makefile
anastasis> config.status: creating src/backend/Makefile
anastasis> config.status: creating src/cli/Makefile
anastasis> config.status: creating src/include/Makefile
anastasis> config.status: creating src/lib/Makefile
anastasis> config.status: creating src/util/Makefile
anastasis> config.status: creating src/reducer/Makefile
anastasis> config.status: creating src/restclient/Makefile
anastasis> config.status: creating src/stasis/Makefile
anastasis> config.status: creating src/testing/Makefile
anastasis> config.status: creating anastasis_config.h
anastasis> config.status: executing depfiles commands
anastasis> config.status: executing libtool commands
anastasis> config.status: executing po-directories commands
anastasis> config.status: creating po/POTFILES
anastasis> config.status: creating po/Makefile
anastasis> building
anastasis> build flags: SHELL=/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash
anastasis> make  all-recursive
anastasis> make[1]: Entering directory '/build/anastasis'
anastasis> Making all in .
anastasis> make[2]: Entering directory '/build/anastasis'
anastasis> make[2]: Leaving directory '/build/anastasis'
anastasis> Making all in contrib
anastasis> make[2]: Entering directory '/build/anastasis/contrib'
anastasis> Making all in .
anastasis> make[3]: Entering directory '/build/anastasis/contrib'
anastasis> ./extract.sh
anastasis> make[3]: Leaving directory '/build/anastasis/contrib'
anastasis> make[2]: Leaving directory '/build/anastasis/contrib'
anastasis> Making all in src
anastasis> make[2]: Entering directory '/build/anastasis/src'
anastasis> Making all in include
anastasis> make[3]: Entering directory '/build/anastasis/src/include'
anastasis> make[3]: *** No rule to make target 'anastasis_error_codes.h', needed by 'all-am'.  Stop.
anastasis> make[3]: Leaving directory '/build/anastasis/src/include'
anastasis> make[2]: *** [Makefile:434: all-recursive] Error 1
anastasis> make[2]: Leaving directory '/build/anastasis/src'
anastasis> make[1]: *** [Makefile:490: all-recursive] Error 1
anastasis> make[1]: Leaving directory '/build/anastasis'
anastasis> make: *** [Makefile:422: all] Error 2
error: builder for '/nix/store/z3lwgi3q7nf6hhv82h15qbcc0i7gp74i-anastasis-0.4.0.drv' failed with exit code 2;
       last 10 log lines:
       > make[2]: Entering directory '/build/anastasis/src'
       > Making all in include
       > make[3]: Entering directory '/build/anastasis/src/include'
       > make[3]: *** No rule to make target 'anastasis_error_codes.h', needed by 'all-am'.  Stop.
       > make[3]: Leaving directory '/build/anastasis/src/include'
       > make[2]: *** [Makefile:434: all-recursive] Error 1
       > make[2]: Leaving directory '/build/anastasis/src'
       > make[1]: *** [Makefile:490: all-recursive] Error 1
       > make[1]: Leaving directory '/build/anastasis'
       > make: *** [Makefile:422: all] Error 2
       For full logs, run 'nix log /nix/store/z3lwgi3q7nf6hhv82h15qbcc0i7gp74i-anastasis-0.4.0.drv'.
error: build of '/nix/store/16bidd516jksj7m4bkh2z9psw5gvr84m-vm-test-run-anastasis-httpd.drv', '/nix/store/z3lwgi3q7nf6hhv82h15qbcc0i7gp74i-anastasis-0.4.0.drv' failed
```
</details>

--- a/pkgs/by-name/anastasis/package.nix
+++ b/pkgs/by-name/anastasis/package.nix
@@ -17,6 +17,7 @@
   gnunet,
   makeWrapper,
   which,
+  callPackage,
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "anastasis";
@@ -80,6 +81,8 @@ stdenv.mkDerivation (finalAttrs: {
     # Check that anastasis-config can find gnunet at runtime
     $out/bin/anastasis-config --help > /dev/null
   '';
+
+  passthru.tests.vmTest = callPackage ./test.nix {};
 
   meta = {
     description = ''

--- a/pkgs/by-name/anastasis/test.nix
+++ b/pkgs/by-name/anastasis/test.nix
@@ -1,0 +1,96 @@
+{
+  lib,
+  nixosTest,
+  anastasis,
+  pkgs,
+  ...
+}:
+nixosTest {
+  name = "anastasis-httpd";
+  nodes = {
+    client = {...}: {
+      systemd.services.anastasis-httpd = {
+        wantedBy = ["multi-user.target"];
+        serviceConfig.ExecStart = ''
+          ${anastasis}/bin/anastasis-httpd
+        '';
+      };
+      environment.systemPackages = with pkgs; [
+        anastasis
+        postgresql
+        taler-exchange
+        taler-merchant
+
+        # needed by src/cli/test_*
+        jq
+        wget
+
+        # needed by make check
+        gnumake
+        automake
+        autoconf
+        autoconf-archive
+        autoconf-archive
+        pkg-config
+        libgcrypt.dev
+        gcc
+      ];
+
+      services.postgresql = {
+        enable = true;
+        initialScript = pkgs.writeText "initialScript.sql" ''
+          create role root login createdb;
+        '';
+      };
+    };
+  };
+  testScript = let
+    check-anastasis = pkgs.writeScript "check-anastasis" ''
+      # Load test fixture data into the vm $HOME (/root)
+      cd ${anastasis.configured} && find . -type f -exec install -Dm 755 "{}" "$HOME/{}" \;
+      cd $HOME
+      # Patch some paths
+      sed=${pkgs.gnused}/bin/sed
+      find . -type f -exec $sed -i "s^/build/source^$HOME^g" "{}" \;
+      find . -type f -exec $sed -i "s^/usr/bin/file^${pkgs.file}/bin/file^g" "{}" \;
+
+      # ./missing is executed at the beginning of `make check` and re-triggers autoreconf
+      #   -> Fix that by making missing a no-op
+      echo ":" > missing
+
+      for i in "" $(seq 1 4); do createdb anastasischeck$i; done
+
+      # Start checking anastasis
+      # FIXME: recursvely adds paths to dependencies
+      # Provide all the paths toward header files and libraries, as well as pkg-config files
+      # This can be debugged by prefixing with "NIX_DEBUG=1 "
+      # FIXME: the build is triggered !!! making the check *very* long :(
+      #  a solution would be to copy the state of the package after build has finished
+      # FIXME: The log of `make check` is only shown at the end
+      NIX_CFLAGS_COMPILE_BEFORE_x86_64_unknown_linux_gnu="${toString (map (p: "-I${lib.getDev p}/include")
+          (with pkgs; [libsodium jansson libgcrypt curl libgnurl libmicrohttpd libtool zlib]))}" \
+        NIX_LDFLAGS_BEFORE_x86_64_unknown_linux_gnu="${toString (map (p: "-L${lib.getLib p}/lib")
+          (with pkgs; [libsodium jansson libgcrypt curl libgnurl libmicrohttpd libtool postgresql libossp_uuid zlib]))}" \
+        PKG_CONFIG_PATH="${lib.concatStringsSep ":" (map (p: "${lib.getDev p}/lib/pkgconfig")
+          (with pkgs; [libmicrohttpd jansson]))}" \
+        make check
+    '';
+  in ''
+    start_all()
+    client.wait_for_unit("multi-user.target")
+
+    print('Copying the fixture and running `make check`')
+    print('The log of `make check` will be shown at the end')
+    print('Wait some long seconds (some postgres ERROR are expected but should probably be fixed) ...')
+    # FIXME: follow the log
+    client.log(client.execute("cd $HOME && set -x && ${check-anastasis}")[1])
+
+    # The interesting part of the log is after "make check_TESTS"
+    # i.e. lines containing "*test_anstasis_*"
+
+    # FIXME:
+    # src/cli/test_anastasis_reducer_enter_secret.sh
+    # and src/cli/recovery_enter_user_attributes.sh
+    # are skipped due to `line {65,64}: taler-bank-manage: command not found`
+  '';
+}

